From 6436a4785c5375b39323f54ef61aadc81f8b70e4 Mon Sep 17 00:00:00 2001
From: jrior001 <jriordan001@gmail.com>
Date: Thu, 24 Nov 2016 22:34:25 -0500
Subject: [PATCH 20/80] Allow lid to send a generic COVER_CHANGED broadcast

This intent will be used to launch a common app for devices
with a view window flip cover(LG, Asus..) or dotcase(HTC).

Updated for Lineage SDK and squashed with the following change:

  Author: Ethan Chen <intervigil@gmail.com>
  Date:   Sat Jan 28 21:39:21 2017 -0800

    Correctly name LID_STATE_CHANGED broadcast

    Change-Id: I472becdcf310fab8536fb24e6cacf646199f464d

Change-Id: I8ce163d0d3f71e40680f42348847e548b41a0045
---
 core/res/AndroidManifest.xml                             | 5 +++++
 .../com/android/server/policy/PhoneWindowManager.java    | 9 +++++++++
 2 files changed, 14 insertions(+)

diff --git a/core/res/AndroidManifest.xml b/core/res/AndroidManifest.xml
index 66c497e9977..16bb90c37fd 100644
--- a/core/res/AndroidManifest.xml
+++ b/core/res/AndroidManifest.xml
@@ -603,6 +603,11 @@
     <protected-broadcast android:name="android.intent.action.DOCK_IDLE" />
     <protected-broadcast android:name="android.intent.action.DOCK_ACTIVE" />
 
+    <!-- LineageOS additions -->
+
+    <!-- Used to launch a common app (FlipFlap) for devices with flip cover. -->
+    <protected-broadcast android:name="lineageos.intent.action.LID_STATE_CHANGED" />
+
     <!-- ====================================================================== -->
     <!--                          RUNTIME PERMISSIONS                           -->
     <!-- ====================================================================== -->
diff --git a/services/core/java/com/android/server/policy/PhoneWindowManager.java b/services/core/java/com/android/server/policy/PhoneWindowManager.java
index 7d6b896b9f0..81e55e2fc2e 100644
--- a/services/core/java/com/android/server/policy/PhoneWindowManager.java
+++ b/services/core/java/com/android/server/policy/PhoneWindowManager.java
@@ -7845,6 +7845,15 @@ public class PhoneWindowManager implements WindowManagerPolicy {
         synchronized (mLock) {
             updateWakeGestureListenerLp();
         }
+        sendLidChangeBroadcast();
+    }
+
+    private void sendLidChangeBroadcast() {
+        Log.d(TAG, "Sending cover change broadcast, mLidState=" + mLidState);
+        Intent intent = new Intent(lineageos.content.Intent.ACTION_LID_STATE_CHANGED);
+        intent.putExtra(lineageos.content.Intent.EXTRA_LID_STATE, mLidState);
+        intent.setFlags(Intent.FLAG_RECEIVER_REPLACE_PENDING);
+        mContext.sendBroadcastAsUser(intent, UserHandle.SYSTEM);
     }
 
     void updateUiMode() {
-- 
2.17.1

